CMakeLists.txt

```diff
@@ -39,6 +39,10 @@ check_c_compiler_flag(-Wno-unused-parameter UV_LINT_NO_UNUSED_PARAMETER)
 check_c_compiler_flag(-Wstrict-prototypes UV_LINT_STRICT_PROTOTYPES)
 check_c_compiler_flag(-Wextra UV_LINT_EXTRA)
 
+if(MSVC AND MSVC_VERSION GREATER 1800)
+    add_definitions(/source-charset:utf-8)
```

Comment:
I'd recommend using a generator expression combined with `add_compile_options`. `add_definitions` has been deprecated for quite some time.
The entire if-block can be replaced with
```cmake
string(CONCAT msvc-utf8 $<
  $<AND:
    $<C_COMPILER_ID:MSVC>
    $<VERSION_GREATER:${MSVC_VERSION},1800>
  >:/source-charset:utf-8
>)
add_compile_options(${msvc-utf8})
```

(I'm aware the generator expression isn't all that readable, but I don't want to

This speeds up configure time and allows generation steps to run faster. It also means that we could eventually add support for the clang-cl variant, which the minimum cmake version doesn't currently support (but may be an issue in the near future as it becomes more and more stable)